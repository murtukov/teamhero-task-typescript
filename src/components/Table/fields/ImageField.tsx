import React from 'react';
import {createUseStyles} from "react-jss";
import {styles as sharedStyles} from "./styles";
import {IFieldProps} from "./sharedProps";

export interface IImageFieldProps extends IFieldProps {
    data?: string,
    shape?: 'circle' | 'rect',
}

function ImageField({data, shape = 'circle'}: IImageFieldProps) {
    const c = useStyles();

    if (undefined === data) {
        return null;
    }

    return (
        <td className={c.root}>
            <img
                src={data}
                alt='avatar'
                width={35}
                height={35}
                className={c.img}
            />
        </td>
    );
}

const useStyles = createUseStyles({
    ...sharedStyles,
    img: {
        borderRadius: 50
    }
});

export default ImageField;