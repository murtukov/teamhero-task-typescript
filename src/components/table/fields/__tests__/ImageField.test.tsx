import React from "react";
import {render} from "@testing-library/react";
import ImageField from "../ImageField";

describe('ImageField component', () => {
    it('matches snapshot', () => {
        const { container } = render(<ImageField data='about:blank' source='' />);
        expect(container.firstChild).toMatchSnapshot();
    });
})