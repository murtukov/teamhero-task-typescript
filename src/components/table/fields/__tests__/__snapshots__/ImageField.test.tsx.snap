// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageField component matches snapshot 1`] = `
<div
  class="cell-0-2-1 cell-d0-0-2-3"
>
  <img
    alt="avatar"
    class="img-0-2-2"
    height="35"
    src="about:blank"
    width="35"
  />
</div>
`;
